storm-shim/src/main/java/storm/mesos/shims/ICommandLineShim.java

```diff
@@ -0,0 +1,12 @@
+package storm.mesos.shims;
+
+/**
+ * Returns the command line, shimmed and 0.9.X and 0.10.X have slightly a different command
+ */
+public interface ICommandLineShim {
+
+  public String getCommandLine();
+
+  public String getDockerCommandLine();
```

Comment:
the interface without ``getDockerCommandLine()` makes more sense to me.  I was expecting to see the following concrete classes:
- `StandardCommandLineShim implements ICommandLineShim`
- `DockerCommandLineShim implements ICommandLineShim`

In the `MesosNimbus` code, you could have:
```java
ICommandLineShim  commandLineShim = Factory.makeCommandLineShim(commandType);
...
  .setCommand(commandLineShim.getCommandLine())
...
```

I think you can avoid having the two separate invocations of `setCommand` with this approach.
```